formatting/models.py

```diff
@@ -11,143 +11,248 @@
+        """Get the absolute URL of the object."""
         return reverse("format:time_detail", kwargs={"pk": self.pk})
 
     class Meta:
         ordering = ("time_id",)
 
 
 class Format(PermissionsBase):
-    """Details of the data file format, combining different aspects."""
-
-    format_id = models.AutoField("format_id", primary_key=True)
+    """Details of the data file format, describing how to read the file.
+
+    It combines several properties, such as the file extension, the delimiter, the date
+    and time formats, and the column indices for the date and time columns, instructing
+    how to read the data file and parse the dates. It is mostly used to ingest data from
+    text files, like CSV.
+
+    Attributes:
+        format_id (AutoField): Primary key.
+        name (CharField): Short name of the format entry.
+        description (TextField): Description of the format.
+        extension (ForeignKey): The extension of the data file.
+        delimiter (ForeignKey): The delimiter between columns in the data file. Only
+            required for text files.
+        first_row (SmallIntegerField): First row of the data, excluding any heading.
+        footer_rows (SmallIntegerField): Number of footer rows, to be ignored at the
+            end.
+        date (ForeignKey): Format for the date column. Only required for text files.
+        date_column (SmallIntegerField): Index of the date column, starting in 1.
+        time (ForeignKey): Format for the time column. Only required for text files.
+        time_column (SmallIntegerField): Index of the time column, starting in 1.
+    """
+
+    format_id = models.AutoField(
+        "format_id", primary_key=True, help_text="Primary key."
+    )
+    name = models.CharField(
+        "Format name", max_length=35, help_text="Short name of the format entry."
+    )
+    description = models.TextField(
+        "Description", blank=True, null=True, help_text="Description of the format."
+    )
     extension = models.ForeignKey(
         Extension,
         on_delete=models.PROTECT,
-        blank=True,
-        null=True,
+        blank=False,
+        null=False,
         verbose_name="File extension",
+        help_text="The extension of the data file.",
     )
     delimiter = models.ForeignKey(
         Delimiter,
         on_delete=models.PROTECT,
         blank=True,
         null=True,
         verbose_name="Delimiter",
+        help_text="The delimiter between columns in the data file. Only required for "
+        "text files.",
     )
-    name = models.CharField("Format name", max_length=35)
-    description = models.TextField("Description", null=True)
-    location = models.CharField("Location", max_length=300, blank=True, null=True)
-    file = models.CharField(
-        "Archivo",
-        max_length=100,
-        blank=True,
-        null=True,
-        help_text="Only applies to automatic transmission",
+    first_row = models.SmallIntegerField(
```

Comment:
Could use [`PositiveSmallIntegerField`](https://docs.djangoproject.com/en/5.1/ref/models/fields/#positivesmallintegerfield)?

Reply:
Indeed. I didn't thought about that. That field still allows `0`, so we still need to do some validation to pick that case.

Reply:
I think I'm going to leave the field as it is - changing to the Positive version does not save adding validators, which would be the main reason for the change.

Reply:
Fair enough. Just wondering - why is the counting from 1 and not zero anyway?